torchvision/datasets/_stereo_matching.py

```diff
@@ -0,0 +1,1041 @@
+                    img1_right.jpg
+                    img1_left.disp.jpg
+                    img1_right.disp.jpg
+                    ...
+                hole
+                    img1_left.jpg
+                    img1_right.jpg
+                    img1_left.disp.jpg
+                    img1_right.disp.jpg
+                    ...
+
+    Args:
+        root (str): Root directory of the dataset.
+        split (str): The split of the dataset to use. One of ``"tree"``, ``"shapenet"``, ``"reflective"``, ``"hole"``
+            or ``"all"``. The ``"all"`` split contains all of the above splits.
+        transforms (callable, optional): A function/transform that takes in a sample and returns a transformed version.
+        download (bool, optional): If true, downloads the dataset from the internet and puts it in the root directory.
+        max_disparity (int, optional): Maximum disparity value. Used to compute the valid mask.
+    """
+
+    DOWNLOAD_SPACE = 400 * 1024 * 1024 * 1024
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "tree",
+        transforms: Optional[Callable] = None,
+        download: bool = False,
+        max_disparity: float = 256.0,
+    ):
+        super().__init__(root, transforms)
+
+        root = Path(root) / "CREStereo"
+        self.max_disparity = max_disparity
+
+        # if the API user requests a dataset download check that the user can download it
+        if download:
+            statvfs = os.statvfs(root)
+            # measured in bytes
+            available_space = statvfs.f_frsize * statvfs.f_bavail
+            if available_space - self.DOWNLOAD_SPACE < 0:
+                raise ValueError(
+                    f"The storage device for {str(root)} is too small to download the dataset), "
+                    f"an additional {self.DOWNLOAD_SPACE - available_space:.2f} GB are required."
+                )
+            self._download_dataset(str(root))
+
+        verify_str_arg(split, "split", valid_values=("tree", "shapenet", "reflective", "hole", "all"))
+
+        splits = {
+            "tree": ["tree"],
+            "shapenet": ["shapenet"],
+            "reflective": ["reflective"],
+            "hole": ["hole"],
+            "all": ["hole", "shapenet", "reflective", "hole"],
+        }[split]
+
+        for s in splits:
+            left_image_pattern = str(root / s / "*_left.jpg")
+            right_image_pattern = str(root / s / "*_right.jpg")
+            imgs = self._scan_pairs(left_image_pattern, right_image_pattern)
+            self._images += imgs
+
+            left_disparity_pattern = str(root / s / "*_left.disp.jpg")
+            right_disparity_pattern = str(root / s / "*_right.disp.jpg")
+            disparities = self._scan_pairs(left_disparity_pattern, right_disparity_pattern)
+            self._disparities += disparities
+
+    def _read_disparity(self, file_path: str) -> Tuple:
+        disparity = np.array(Image.open(file_path), dtype=np.float32)
```

Comment:
Here an everywhere else where there is a similar use of `np.array(pil_img)`: we should try to use `np.asarray` so that the input is only copied when needed